plugin-template

```diff
@@ -0,0 +1,269 @@
+    parser.add_argument('--generate-config', action='store_true',
+                        help=textwrap.dedent('''\
+                            Create or update a plugin template config for a plugin and exit.
+
+                        '''))
+    parser.add_argument('--bootstrap', action='store_true',
+                        help=textwrap.dedent('''\
+                            Create a new plugin and template boilerplate code.
+
+                        '''))
+    parser.add_argument('--test', action='store_true',
+                        help=textwrap.dedent('''\
+                            Generate or update functional and unit tests.
+
+                        '''))
+    parser.add_argument('--travis', action='store_true',
+                        help=textwrap.dedent('''\
+                            Generate or update travis configuration files.
+
+                        '''))
+    parser.add_argument('--docs', action='store_true',
+                        help=textwrap.dedent('''\
+                            Generate or update plugin documentation.
+
+                        '''))
+    parser.add_argument('--all', action='store_true',
+                        help=textwrap.dedent('''\
+                            Create a new plugin and template all non-excluded files.
+
+                        '''))
+    parser.add_argument('--verbose', action='store_true',
+                        help=textwrap.dedent('''\
+                            Include more output.
+
+                        '''))
+    args = parser.parse_args()
+
+    config = generate_config(args.plugin_name, args.plugin_app_label)
+
+    here = os.path.dirname(os.path.abspath(__file__))
+    plugin_root_dir = os.path.join(os.path.dirname(here), args.plugin_name)
+    plugin_config_path = os.path.join(plugin_root_dir, 'template_config.yml')
+
+    try:
+        with open(plugin_config_path) as config_file:
+            try:
+                config_in_file = yaml.safe_load(config_file)
+                if config_in_file:
+                    # TODO: validate config
+                    config = config_in_file
+                    # Add any missing value from the list of defaults
+                    for key, value in DEFAULT_SETTINGS.items():
+                        if key not in config:
+                            config[key] = value
+                    print("\nLoaded plugin template config from "
+                          "{path}/template_config.yml.\n".format(path=plugin_root_dir))
+                    if args.verbose:
+                        print("\nUsing the following config:\n")
+                        pp = pprint.PrettyPrinter(indent=4)
+                        pp.pprint(config)
+                        print("\n")
+            except yaml.YAMLError as exc:
+                print(exc)
+                exit()
+    except FileNotFoundError:
+        print("\nCould not find a plugin template config at {path}/template_config.yml.\n".format(
```

Comment:
Proposal: Add `Using default configuration` or so to make clear that this is not an error.

Reply:
I'll add that language.

Reply:
Optional: Could the exception mention  --generate config?